### PR TITLE
add automated builds with jenkins

### LINKED ISSUE #1: Image build should use git for determining image tag
Should protect against accidentally overwriting existing image tags (need to remember to update tag in the build script). Instead could just get this info from git.

Could be tag or hash. One example from bootkube: https://github.com/kubernetes-incubator/bootkube/blob/master/image/bootkube/build-image.sh#L7

### PR DESCRIPTION
**Pull request by abhinavdahiya**
Add automated builds on master branch.
closes #1 
/cc: @aaronlevy @yifan-gu

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+      }
+      environment {
+        QUAY = credentials('174f0169-803f-4840-8e07-b8bfa204f5c6')
+      }
+      steps {
+        sh """
+        cat /etc/os-release
+        docker version
+        docker login -u=$QUAY_USR -p=$QUAY_PSW quay.io
+        export CNI_VERSION=v0.5.2
```
Comment (aaronlevy) [inline Jenkinsfile]: nit: Could this be sourced from a single location? Having it stored multiple times means it could get out of sync

Comment (abhinavdahiya) [inline Jenkinsfile]: should I enforce CNI_VERSION in scripts/build-image only?

Comment (aaronlevy) [inline Jenkinsfile]: That might be ideal - given that is essentially the entry-point to what we are actually releasing

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+     label 'coreos && !rkt-fly'
+  }
+
+  stages {
+    stage('build-image') {
+      when {
+        branch 'master'
+      }
+      environment {
+        QUAY = credentials('174f0169-803f-4840-8e07-b8bfa204f5c6')
```
Comment (dghubble) [inline Jenkinsfile]: you can give these meaningful names

**Comment by abhinavdahiya** [comment]
ping @aaronlevy @yifan-gu

File: Makefile [hunk 3]
```diff
@@ -1,12 +1,11 @@
-CURL=curl -sSf
+CURL:=curl -sSf
 
-IMAGE_NAME := quay.io/coreos/flannel-cni
-VERSION ?= latest
-TAG ?= 0.1
-CNI_VERSION ?= v0.5.2
+FLANNEL_CNI_ROOT:=$(shell git rev-parse --show-toplevel)
+IMAGE_NAME:=quay.io/coreos/flannel-cni
+VERSION:=$(shell $FLANNEL_CNI_ROOT/scripts/git-version)
```
Comment (dghubble) [inline Makefile]: why are there two different versions? The aim of the git-version script is to automatically read the version as a Git SHA hash or Git tag (if present). Seems like you'd either want to use that tagging pattern or set the version directly, not both.

Comment (abhinavdahiya) [inline Makefile]: > why are there two different versions?
which are two different versions...

Comment (dghubble) [inline Makefile]: You have VERSION and a hardcoded CNI_VERSION

Comment (dghubble) [inline Makefile]: I guess you're not planning to match them up and only want to use it for curl's. Disregard

Comment (abhinavdahiya) [inline Makefile]: I don't think we should tie CNI's release version with flannel-cni version, hence they are separate.

File: scripts/build-image.sh [hunk 4]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
+set -e
+CURL="curl -sSf"
+FLANNEL_CNI_ROOT=$(git rev-parse --show-toplevel)
+IMAGE_NAME=quay.io/coreos/flannel-cni
+VERSION=$($FLANNEL_CNI_ROOT/scripts/git-version)
+CNI_VERSION="v0.5.2"
```
Comment (dghubble) [inline scripts/build-image.sh]: Are you expecting users to configure this via environment variables?

Comment (abhinavdahiya) [inline scripts/build-image.sh]: https://github.com/coreos/flannel-cni/pull/2#discussion_r130934208

Comment (dghubble) [inline scripts/build-image.sh]: His original comment still applies, v0.5.2 appears in two places.

Comment (dghubble) [inline scripts/build-image.sh]: don't stress it, as long as the maintainers are cool with this approach to scripting

Comment (aaronlevy) [inline scripts/build-image.sh]: Hmmm - this does seem like the same process as in the makefile, no? Any reason we need to the makefile and the build-image script?

Comment (abhinavdahiya) [inline scripts/build-image.sh]: I didn't remove the Makefile just for local dev, but the image on jenkins only uses build-image.sh
I totally fine with removing the makefile altogether. @aaronlevy

Comment (aaronlevy) [inline scripts/build-image.sh]: You could have the makefile just run the bash script... kinda up to you. Wouldn't block either way -- just potential later friction to have version defined in multiple places.

Comment (abhinavdahiya) [inline scripts/build-image.sh]: @aaronlevy done

**Pull request merged**